### PR TITLE
google_rtc_audio_processing: Add config controls

### PR DESCRIPTION
**Pull request by lkoenig**
1. Added a byte control to accept Google RTC audio processing
   configuration blob
     `name='GOOGLE_RTC_PROCESSING10.0 Config'`

Tested using:
    sof-ctl -n 48 -t 0 -b -r -s /root/tuning.bin
And verified correct audio processing behaviour on a alderlake board.

Signed-off-by: Ben Zhang <benzh@chromium.org>
Signed-off-by: Lionel Koenig <lionelk@google.com>

### CODE REVIEW

File: src/audio/google_rtc_audio_processing.c [hunk 1]
```diff
@@ -83,6 +83,35 @@ static int google_rtc_audio_processing_params(
...
+	uint8_t *config;
+	size_t size;
+	int ret;
+	size_t i;
+
+	comp_info(dev, "google_rtc_audio_processing_reconfigure()");
+
+	config = comp_get_data_blob(cd->tuning_handler, &size, NULL);
+	if (!config || !size) {
+		comp_err(dev, "google_rtc_audio_processing_reconfigure(): Tuning config not set");
```
Comment (cujomalainey) [inline src/audio/google_rtc_audio_processing.c:98]: you emit an error but return 0

**Review by cujomalainey — CHANGES_REQUESTED** [review]

File: src/audio/google_rtc_audio_processing.c (hunk 1 above)
Comment (lkoenig) [reply src/audio/google_rtc_audio_processing.c]: Indeed.
I splitted in 2 case:
  One where we do not have config data and silently exit.
  One wherewe should have some and get no pointer.

**Comment by lkoenig** [comment]
Codestyle issue are from an internal header file I want to keep as close as possible to internal code style.

**Review by lgirdwood — APPROVED** [review]

**Comment by lkoenig** [comment]
@cujomalainey Do you require more change ?

**Comment by cujomalainey** [comment]
Checkpatch appears to be complaining about tabs v spaces and some other spacing stuff

**Review by cujomalainey — APPROVED** [review]

**Comment by cujomalainey** [comment]
Failures are either not relevant or being ignored (checkpatch) as the file is imported

**Pull request merged**